### DISCUSSION TITLE
leafletObject isn't defined..

### DISCUSSION THREAD

**Discussion by q3dm15** [vue-leaflet/vue-leaflet#345] [General]
```
mounted () {
    this.$nextTick( async () => {
      this.map = this.$refs.refMap.leafletObject
      console.log(this.map.leafletObject) // this's undefined variable, why?
      const response = await axios.post('./public/data/pts.geojson')
      this.geoJson.push({geojson: response.data})
    })
  },
```
ref defined but vue 3 can't see leafletObject. How I can interactive with `refMap` as with `L.map('refMap')`
```
  <l-map :use-global-leaflet="false" ref="refMap" :options="optionsMap" />
```

**Comment by Soulfarmer** [comment]
i was able to access it using
`var map = this.$refs.refMap as typeof LMap`;

**Reply by q3dm15** [reply]
I don't use TS

**Reply by q3dm15** [reply]
this's given value 
![image](https://github.com/vue-leaflet/vue-leaflet/assets/133960645/15970a69-6e63-4154-9a8b-f158555f411f)

**Reply by q3dm15** [reply]
and I tried to cast value, so JSX don't support it `console.log(<LMap> __self.map)`
`JSX element 'LMap' has no corresponding closing tag.ts(17008)`

**Comment by q3dm15** [comment]
hmm.. it works after timer delay
or with ...
```
methods: {
  onMapReady() {
    this.map = this.$refs.refMap.leafletObject
  }
}
```

**Reply by Soulfarmer** [reply]
Someone said on a different thread that they also got it working with the $nextTick function.
```
 mounted() {
    this.$nextTick(function () {
    his.map = this.$refs.refMap.leafletObject
    })
  },
  ```
Might interest you also.
Can't find it again unfortunatly, i cant give proper credit.

**Reply by q3dm15** [reply]
because their logics somewhere has time delay
